text/0000-default-and-expanded-rustc-errors.md

```diff
@@ -0,0 +1,290 @@
+
+Secondary labels help to understand the error and use blue text and --- underline.  These labels explain the **why** of the compiler message.  You can see one such example in the above message where the secondary labels explain that there is already another borrow going on.  In another example, we see another way that primary and secondary work together to tell the whole story for why the error occurred.
+
+Taken together, primary and secondary labels create a 'flow' to the message.  Flow in the message lets the user glance at the colored labels and quickly form an educated guess as to how to correctly update their code.
+
+Note: We'll talk more about additional style guidance for wording to help create flow in the subsequent style RFC. 
+
+## Expanded error messages
+
+Currently, --explain text focuses on the error code.  You invoke the compiler with --explain <error code> and receive a verbose description of what causes errors of that number.  The resulting message can be helpful, but it uses generic sample code which makes it feel less connected to the user's code.
+
+We propose adding a new `--explain errors`.  By passing this to the compiler (or to cargo), the compiler will switch to an expanded error form which incorporates the same source and label information the user saw in the default message with more explanation text.
+
+```
+error: cannot move out of borrowed content
+   --> /Users/jturner/Source/errors/borrowck-move-out-of-vec-tail.rs:30:17
+
+I’m trying to track the ownership of the contents of `tail`, which is borrowed, through this match statement:
+
+29  |              match tail {
+
+In this match, you use an expression of the form [...].  When you do this, it’s like you are opening up the `tail` 
+value and taking out its contents. Because `tail` is borrowed, you can’t safely move the contents.
+
+30  |                  [Foo { string: aa },
+    |                                 ^^ cannot move out of borrowed content
+
+You can avoid moving the contents out by working with each part using a reference rather than a move.  A naive fix 
+might look this:
+
+30  |                  [Foo { string: ref aa },
+```
+
+*Example of an expanded error message*
+
+The expanded error message effectively becomes a template.  The text of the template is the educational text that is explaining the message more more detail.  The template is then populated using the source lines, labels, and spans from the same compiler message that's printed in the default mode.  This lets the message writer call out each label or span as appropriate in the expanded text.
+
+It's possible to also add additional labels that aren't necessarily shown in the default error mode but would be available in the expanded error format.  This gives the explain text writer maximal flexibility without impacting the readability of the default message.  I'm currently prototyping an implementation of how this templating could work in practice.
+
```

Comment:
I'm not sure I follow this point.  The templated language piece seems more engineer-related rather than design related.  We could iterate the templating piece without impacting the user's experience.